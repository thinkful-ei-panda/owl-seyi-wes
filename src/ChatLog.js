import React from 'react';
import ChatEvent from './ChatEvent';

function ChatLog(props){
    return (
        <section className='chat-log'>
        {
        props.chatEvents.map((arrEle,index) => <ChatEvent key={index} participants={props.participants} chatEvent={props.chatEvents[index]} />)
        }
        </section>
        
    )
}

export default ChatLog