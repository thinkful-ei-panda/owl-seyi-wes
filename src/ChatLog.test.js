import React from 'react';
import ReactDOM from 'react-dom';
import ChatLog from './ChatLog';
import store from './store'

import renderer from 'react-test-renderer';

//<ChatLog chatEvents={props.chatEvents} participants={props.participants}/>
describe('ChatLog', () => {

    const props={
        chatEvents:store.chatEvents,
        participants:store.participants
    }
  it('renders without crashing', () => {

    const div = document.createElement('div');
    ReactDOM.render(<ChatLog {...props}/>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<ChatLog {...props}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });

});

