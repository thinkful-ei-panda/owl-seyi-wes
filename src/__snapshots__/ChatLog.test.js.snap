// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatLog renders the UI as expected 1`] = `
<section
  className="chat-log"
>
  <div
    className="chat-event-container"
  >
    <div
      className="small"
    >
      <img
        alt="avatar"
        src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
      />
    </div>
    <div>
      <div>
        Axe Kubicka
        <span
          className="time"
        >
          4:48 AM
        </span>
      </div>
      <div>
        Hello world
      </div>
    </div>
  </div>
  <div
    className="chat-event-container2 center"
  >
    <div>
      4:48 AM
    </div>
    <div
      className="event"
    >
      Caty Flucker
       gave a thumbs up
    </div>
  </div>
  <div
    className="chat-event-container2 center"
  >
    <div>
      4:48 AM
    </div>
    <div
      className="event"
    >
      Frank Runciman
       gave a thumbs down
    </div>
  </div>
  <div
    className="chat-event-container2 center"
  >
    <div>
      4:49 AM
    </div>
    <div
      className="event"
    >
      Koren Templeton
       raised their hand
    </div>
  </div>
  <div
    className="chat-event-container2 center"
  >
    <div>
      4:49 AM
    </div>
    <div
      className="event"
    >
      Koren Templeton
       clapped
    </div>
  </div>
  <div
    className="chat-event-container2 center"
  >
    <div>
      4:49 AM
    </div>
    <div
      className="event"
    >
      Koren Templeton
       joined
    </div>
  </div>
  <div
    className="chat-event-container2 center"
  >
    <div>
      4:50 AM
    </div>
    <div
      className="event"
    >
      Ashla Attwool
       left the session
    </div>
  </div>
  <div
    className="chat-event-container2 center"
  >
    <div>
      4:51 AM
    </div>
    <div
      className="event"
    >
      Axe Kubicka
       joined the stage
    </div>
  </div>
  <div
    className="chat-event-container2 center"
  >
    <div>
      4:52 AM
    </div>
    <div
      className="event"
    >
      Axe Kubicka
       left the stage
    </div>
  </div>
</section>
`;
